render/src/render.rs

```diff
@@ -3,13 +3,16 @@ use std::fmt::{Result, Write};
 /// Render a component
 ///
 /// This is the underlying mechanism of the `#[component]` macro
-pub trait Render: Sized {
+pub trait Render {
     /// Render the component to a writer.
     /// Make sure you escape html correctly using the `render::html_escaping` module
     fn render_into<W: Write>(self, writer: &mut W) -> Result;
 
     /// Render the component to string
-    fn render(self) -> String {
+    fn render(self) -> String
+    where
+        Self: Sized,
+    {
         let mut buf = String::new();
         self.render_into(&mut buf).unwrap();
```

Comment:
This stuff makes the Render trait more portable-- we still need sized as its a dependency of the writer structure we're doing, but we can at least scope this requirement to the render method so that it'll only complain about being passed around when used directly in the html! macro... 🤞